WordPress/src/main/java/org/wordpress/android/GCMMessageService.java

```diff
@@ -425,4 +435,48 @@ public static void removeAllNotifications(Context context) {
 
         clearNotifications();
     }
+
+    //NoteID is the ID if the note in WordPress
+    public static void bumpPushNotificationsTappedAnalytics(String noteID) {
+        for (int id : mActiveNotificationsMap.keySet()) {
+            Bundle noteBundle = mActiveNotificationsMap.get(id);
+            if (noteBundle.getString(PUSH_ARG_NOTE_ID, "").equals(noteID)) {
+                bumpPushNotificationsAnalytics(Stat.PUSH_NOTIFICATION_TAPPED, noteBundle, null);
+                AnalyticsTracker.flush();
+                return;
+            }
+        }
+    }
+
+    // Mark all notifications as tapped
+    public static void bumpPushNotificationsTappedAllAnalytics() {
+        for (int id : mActiveNotificationsMap.keySet()) {
+            Bundle noteBundle = mActiveNotificationsMap.get(id);
+            bumpPushNotificationsAnalytics(Stat.PUSH_NOTIFICATION_TAPPED, noteBundle, null);
```

Comment:
They are just enqueued individually. Sent to the server 2 lines later, with one network call, when calling the `flush` method.

Unfortunately when grouped we need to mark all of them as "tapped", since there is no clear way to get which ones the user is interested in.